src/Exceptions/NotImplementedException.php

```diff
@@ -0,0 +1,13 @@
+<?php
+
+namespace Microsoft\Kiota\Serialization\Multipart\Exceptions;
+
+use BadMethodCallException;
+
+class NotImplementedException extends BadMethodCallException
```

Comment:
Would this be better placed in abstractions? Or do we not forsee the use of it anywhere else?